scarb/tests/fmt.rs

```diff
@@ -270,3 +303,77 @@ fn workspace_with_root() {
     let content = fs::read_to_string(t.child("second/src/lib.cairo")).unwrap();
     assert_eq!(content, SIMPLE_FORMATTED);
 }
+
+#[test]
+fn workspace_emit_with_root() {
+    let t = TempDir::new().unwrap().child("test_workspace");
+    let pkg1 = t.child("first");
+    ProjectBuilder::start()
+        .name("first")
+        .lib_cairo(SIMPLE_ORIGINAL)
+        .build(&pkg1);
+    let pkg2 = t.child("second");
+    ProjectBuilder::start()
+        .name("second")
+        .lib_cairo(SIMPLE_ORIGINAL)
+        .dep("first", &pkg1)
+        .build(&pkg2);
+    let root = ProjectBuilder::start()
+        .name("some_root")
+        .lib_cairo(SIMPLE_ORIGINAL)
+        .dep("first", &pkg1)
+        .dep("second", &pkg2);
+    WorkspaceBuilder::start()
+        .add_member("first")
+        .add_member("second")
+        .package(root)
+        .build(&t);
+
+    Scarb::quick_snapbox()
+        .arg("fmt")
+        .arg("--emit")
+        .arg("stdout")
+        .current_dir(&t)
+        .assert()
+        .failure()
+        .stdout_matches(indoc! {"\
+            fn main() -> felt252 {
```

Comment:
Added a test,
It's unclear to me why I can't pass `SIMPLE_FORMATTED` here, for lack of a newline at end of line

Reply:
It just stems from usage of `println` internally by Scarb ui https://github.com/software-mansion/scarb/blob/333cdb048eab98d5d331aadc16fcd78c25a4985e/utils/scarb-ui/src/message.rs#L33C13-L33C20

You can use `format!("{SIMPLE_FORMATTED}\n")` to account for that here.

Reply:
Thanks! Fixed